haystack/components/converters/xlsx.py

```diff
@@ -0,0 +1,174 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import io
+from pathlib import Path
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
+
+import pandas as pd
+
+from haystack import Document, component, logging
+from haystack.components.converters.utils import get_bytestream_from_source, normalize_metadata
+from haystack.dataclasses import ByteStream
+from haystack.lazy_imports import LazyImport
+
+logger = logging.getLogger(__name__)
+
+with LazyImport("Run 'pip install openpyxl'") as xlsx_import:
+    import openpyxl  # pylint: disable=unused-import # the library is used but not directly referenced
+
+with LazyImport("Run 'pip install tabulate'") as tabulate_import:
+    from tabulate import tabulate  # pylint: disable=unused-import # the library is used but not directly referenced
+
+
+@component
+class XLSXToDocument:
+    """
+    Converts XLSX files to Documents.
+
+    By default, it reads all work sheets into CSV format.
```

Comment:
I would add some additional information.
For example: each sheet is converted into a Document, etc.

Reply:
Let me know if I should add anymore!

Reply:
perfect